internal/domain/product.go

```diff
@@ -0,0 +1,60 @@
+package domain
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+)
+
+var ProductCategory = []string{"Clothing", "Accescories", "Footware", "Beverages"}
+
+type Product struct {
+	Id          string    `json:"id" db:"id"`
+	CreatedAt   time.Time `json:"createdAt" db:"created_at"`
+	Name        string    `json:"name" db:"name"`
+	Sku         string    `json:"sku" db:"sku"`
+	Category    string    `json:"category" db:"category"`
+	ImageUrls   []string  `json:"imageUrls" db:"image_urls"`
+	Notes       string    `json:"notes" db:"notes"`
+	Price       int       `json:"price" db:"price"`
+	Stock       int       `json:"stock" db:"stock"`
+	Location    string    `json:"location" db:"stock"`
+	IsAvailable bool      `json:"isAvailable" db:"is_available"`
+}
+
+type CreateProduct struct {
+	Name        string   `json:"name"`
+	Sku         string   `json:"sku"`
+	Category    string   `json:"category"`
+	ImageUrls   []string `json:"imageUrls"`
+	Notes       string   `json:"notes"`
+	Price       int      `json:"price"`
+	Stock       int      `json:"stock"`
+	Location    string   `json:"location"`
+	IsAvailable bool     `json:"isAvailable"`
+}
+
+type CreateProductResponse struct {
+	Id        string    `json:"id"`
+	CreatedAt time.Time `json:"createdAt"`
+}
+
+func NewProductFromRequestBody(body CreateProduct) Product {
```

Comment:
gin punya function mapping tersendiri. pakai shouldBinJSON

https://gin-gonic.com/docs/examples/binding-and-validation/

Reply:
fungsi ini utk nge-generate id sama created_at utk disimpen di db + merge value dri request body

Reply:
utk mapping request, udh ada tersendiri

Reply:
oh ok2. penamaannya mar. lebih baik fromDTO jg req body. req body identik sama request dari httpnya

Reply:
done